dmodel/dm-article.h

```diff
@@ -37,6 +37,12 @@ DM_AVAILABLE_IN_ALL
 GVariant *
 dm_article_get_table_of_contents (DmArticle *self);
 
+DM_AVAILABLE_IN_ALL
+GInputStream *
+dm_article_get_archive_member_content_stream (DmArticle *self,
+                                              const char *member_name,
+                                              GError **error);
+
```

Comment:
This public interface makes me a little uneasy. Did you consider adding a separate DmArchive content type? Could inherit from either DmContent or DmArticle. We'd need our object metadata (as used in `dm_model_from_json_node`) to specify a different type for archives, but then we'd be able to add a separate DmArchive content type. DmArchive could inherit from either DmContent or DmArticle - whichever makes sense. I think that would make this a bit more natural to use from other places.

Reply:
That would not actually follow the convention, since we use the same content type for PDFs and HTML besides they are substantially different kind of files, I kind of used the same logic for this...

Reply:
Oooh, I see that now. Cool beans! The meat of this looks good to me :) Perhaps we can wiggle things around in the future.